src/api.ts

```diff
@@ -1,19 +1,58 @@
+// @ts-nocheck There are some issues in the Definitely Typed packages of the "ref" related dependencies.
 import { INCHIAPI } from "./ffis";
+import { inchi_OutputStruct, inchi_OutputStructEx } from "./headers";
 
-export function CheckINCHIKey(input: string): 0 | -1 | 1 | 2 | 3 {
-  return INCHIAPI.CheckINCHIKey(input) as 0 | -1 | 1 | 2 | 3;
-}
+// #region Types and Interfaces
 
-export function CheckINCHI(input: string, strict?: boolean): 0 | -1 | 1 | 2 | 3 | 4 {
-  return INCHIAPI.CheckINCHI(input, strict === true ? 1 : 0) as 0 | -1 | 1 | 2 | 3 | 4;
-}
+export type StatusReturnType1 = -1 | 0 | 1 | 2 | 3;
+export type StatusReturnType2 = -1 | 0 | 1 | 2 | 3 | 4;
 
-export function GetStringLength(input: string): number {
-  return INCHIAPI.GetStringLength(input);
-}
+export const GetINCHIReturnCode = {
+  /**
+   * Success; no errors or warnings
```

Comment:
Go to the previous version of the PR and address all the comments made there by bringing change in this PR.

Reply:
sure sir